### ISSUE TITLE
Investigate code oddity

### ISSUE THREAD

**Issue by aeoncleanse** [FAForever/fa#1759]
```
    DetachCargo = function(self)
        local units = self:GetCargo()
        for k, v in units do
            if EntityCategoryContains(categories.TRANSPORTATION, v) then
                for k, u in self:GetCargo() do
                    u:Kill()
                end
            end
            v:DetachFrom()
        end
    end
```
Found in defaultunits.lua in the base transport class. I think the second self:GetCargo() should be v:GetCargo(). Can someone track this code back to see when it was put in and why?

**Comment by aeoncleanse** [comment]
According to @Exotic-Retard due to black box engine magic, by the time this is called by OnKilled for air transports, units will always return an empty table... Needs confirming.

**Comment by HUSSAR-mtrela** [comment]
Correct me if I'm wrong here, but it looks like this code is supposed to kill all unis that are attached the transport unit (in that case self) so why there is a need for nested 2nd for loop? If the reason for 2nd for loop is to find the transport unit then there should not be two function calls to self:GetCargo() but rather this code should look like:
```
DetachCargo = function(self)
        local units = self:GetCargo()
        for k, v in units do
            if EntityCategoryContains(categories.TRANSPORTATION, v) then
                for k, u in v:GetCargo() do
                    u:Kill()
                end
            end
            v:DetachFrom()
       end
end
```
note there is `v:GetCargo() `instead of `self:GetCargo() `in the 2nd for loop. 

I wonder if we can simplify this function to just this code:

```
DetachCargo = function(self)
    if EntityCategoryContains(categories.TRANSPORTATION, self) then
        for k, u in self:GetCargo() do
            u:Kill()
        end
        self:DetachFrom()
    end
end
```

**Comment by aeoncleanse** [comment]
Fixed in fb0aff30cb0cfc568569a3397f3a74ef6d418bd2
